drivers/spi/spi_cc23x0.c

```diff
@@ -0,0 +1,349 @@
+		LOG_ERR("Frequencies lower than %d Hz are not supported", SPI_CC23_MIN_FREQ);
+		return -EINVAL;
+	}
+
+	if (2 * config->frequency > CPU_FREQ) {
+		LOG_ERR("Frequency greater than supported in master mode");
+		return -EINVAL;
+	}
+
+	if (SPI_MODE_GET(config->operation) & SPI_MODE_CPOL) {
+		if (SPI_MODE_GET(config->operation) & SPI_MODE_CPHA) {
+			protocol = SPI_FRF_MOTO_MODE_7;
+		} else {
+			protocol = SPI_FRF_MOTO_MODE_6;
+		}
+	} else {
+		if (SPI_MODE_GET(config->operation) & SPI_MODE_CPHA) {
+			protocol = SPI_FRF_MOTO_MODE_5;
+		} else {
+			protocol = SPI_FRF_MOTO_MODE_4;
+		}
+	}
+
+	/* Enable clock */
+	CLKCTLEnable(CLKCTL_BASE, CLKCTL_SPI0);
+
+	/* Disable SPI before making configuration changes */
+	SPIDisable(cfg->base);
+
+	/* Configure SPI */
+	SPIConfigSetExpClk(cfg->base, CPU_FREQ, protocol, SPI_MODE_CONTROLLER,
+			   config->frequency, SPI_CC23_DATA_WIDTH);
+
+	data->ctx.config = config;
+
+	/* Configure Rx FIFO level */
+	HWREG(cfg->base + SPI_O_IFLS) = SPI_IFLS_RXSEL_LEVEL_1;
+
+	/* Re-enable SPI after making configuration changes */
+	SPIEnable(cfg->base);
+
+	return 0;
+}
+
+static void spi_cc23x0_initialize_data(struct spi_cc23x0_data *data)
+{
+	data->tx_len_left = MAX(spi_context_total_tx_len(&data->ctx),
+				spi_context_total_rx_len(&data->ctx));
+	data->tx_count = 0;
+	data->rx_count = 0;
+	data->xfer_done = false;
+}
+
+static int spi_cc23x0_transceive(const struct device *dev,
+				 const struct spi_config *config,
+				 const struct spi_buf_set *tx_bufs,
+				 const struct spi_buf_set *rx_bufs)
+{
+	const struct spi_cc23x0_config *cfg = dev->config;
+	struct spi_cc23x0_data *data = dev->data;
+	struct spi_context *ctx = &data->ctx;
+	int ret;
+
+	spi_context_lock(ctx, false, NULL, NULL, config);
+
+	ret = spi_cc23x0_configure(dev, config);
+	if (ret) {
+		goto ctx_release;
+	}
+
+	spi_context_buffers_setup(ctx, tx_bufs, rx_bufs, SPI_CC23_DFS);
+
+	spi_cc23x0_initialize_data(data);
+
+	spi_context_cs_control(ctx, true);
+
+	SPIEnableInt(cfg->base, SPI_CC23_INT_MASK);
+
+	ret = spi_context_wait_for_completion(ctx);
+	if (ret) {
+		SPIDisableInt(cfg->base, SPI_CC23_INT_MASK);
+		LOG_ERR("SPI transfer failed (%d)", ret);
+	} else {
+		LOG_DBG("SPI transfer completed");
+	}
+
+	spi_context_cs_control(ctx, false);
+
+ctx_release:
+	spi_context_release(ctx, ret);
+	return ret;
+}
+
+static int spi_cc23x0_release(const struct device *dev,
+			      const struct spi_config *config)
+{
+	const struct spi_cc23x0_config *cfg = dev->config;
+	struct spi_cc23x0_data *data = dev->data;
+
+	if (!spi_context_configured(&data->ctx, config)) {
+		return -EINVAL;
+	}
+
+	if (SPIBusy(cfg->base)) {
+		return -EBUSY;
+	}
+
+	spi_context_unlock_unconditionally(&data->ctx);
+
+	return 0;
+}
+
+static const struct spi_driver_api spi_cc23x0_driver_api = {
+	.transceive = spi_cc23x0_transceive,
+	.release = spi_cc23x0_release,
+};
+
+#define SPI_CC23X0_INIT_FUNC(n)							\
+	static int spi_cc23x0_init_##n(const struct device *dev)		\
```

Comment:
bit sad to generate the whole init as a macro, can you do it only for the interrupt? plenty of examples in the other drivers, grep for `irq_config_func`

then it should become compact enough that you would not bother having a dedicated macro for it and can unwind this into `SPI_CC23X0_INIT`